test/WebSites/RazorPagesWebSite/CustomActionResultModel.cs

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+
+namespace RazorPagesWebSite
+{
+    public class CustomActionResultModel : PageModel
+    {
+        public string MethodName { get; set; }
+
+        public IActionResult OnGet()
+        {
+            return View();
+        }
+
+        public async Task<IActionResult> OnPostAsync()
+        {
+            await Task.Delay(1);
+            MethodName = nameof(OnPostAsync);
+            return View();
+        }
+
+        public async Task OnGetCustomer()
+        {
+            await Task.Delay(1);
+            MethodName = nameof(OnGetCustomer);
+        }
+
+        public async Task OnGetViewCustomerAsync()
+        {
+            await Task.Delay(1);
```

Comment:
Eww, just have these return `TaskCache.CompletedTask`. Shouldn't use delays unless you need them.